src/middleware/nasc.ts

```diff
@@ -108,13 +115,56 @@ async function NASCMiddleware(request: express.Request, response: express.Respon
 		if (pid) {
 			const linkedPIDs: number[] = device.linked_pids;
 
+			// * If a user performs a system transfer from
+			// * a console to another using a Nintendo account
+			// * during the transfer and both consoles have
+			// * a Pretendo account, the devices will be swapped
+			// * since we check it using the LFCS.
+			// *
+			// * So, the linked PIDs won't have the user's PID
+			// * anymore.
 			if (!linkedPIDs.includes(pid)) {
-				response.status(200).send(nascError('102').toString());
-				return;
+				const session: mongoose.ClientSession = await databaseConnection().startSession();
+				await session.startTransaction();
+
+				device.linked_pids.push(pid);
+
+				await device.save({ session });
+
+				await session.commitTransaction();
 			}
 		}
 	}
 
+	// * Workaround for edge case on system transfers
+	// * if a console that has a Pretendo account performs
+	// * a system transfer using the Nintendo account to
+	// * another that doesn't have a Pretendo account.
+	// *
+	// * This would make the Pretendo account to not have
+	// * a device on the database.
+	// *
+	// * TODO: With this change, now multiple devices can
+	// * have the same serial number and MAC address.
+	// * Do we want this? If not, are there other solutions?
```

Comment:
No I do not believe we want this, as it makes it much harder to differentiate between unique consoles. Does a system transfer also move the serial number?

Reply:
I was going to suggest about searching the device using the serial number and MAC address, but since the serial number is also transfered, this puts us on a tight spot.